packages/server/tests/acceptance/rpc_batch2.spec.ts

```diff
@@ -989,4 +993,105 @@ describe('@api-batch-2 RPC Server Acceptance Tests', function () {
   });
+
+  describe('Formats of addresses in Transaction and Receipt results', () => {
+    const getTxData = async (hash) => {
+      const txByHash = await relay.call(RelayCalls.ETH_ENDPOINTS.ETH_GET_TRANSACTION_BY_HASH, [hash], requestId);
+      const receipt = await relay.call(RelayCalls.ETH_ENDPOINTS.ETH_GET_TRANSACTION_RECEIPT, [hash], requestId);
+      let mirrorResult = await mirrorNode.get(`/contracts/results/${hash}`, requestId);
+
+      return { txByHash, receipt, mirrorResult };
+    };
+
+    it('from/to Addresses in transaction between accounts are in evm format', async function () {
+      const tx = await accounts[0].wallet.sendTransaction({
+        to: accounts[1].wallet,
+        value: ethers.parseEther('1'),
+      });
+
+      await tx.wait();
+
+      let { txByHash, receipt, mirrorResult } = await getTxData(tx.hash);
+
+      mirrorResult.from = accounts[0].wallet.address;
+      mirrorResult.to = accounts[1].wallet.address;
+
+      Assertions.transaction(txByHash, mirrorResult);
+      Assertions.transactionReceipt(receipt, mirrorResult);
+
+      Assertions.evmAddress(txByHash.from);
+      Assertions.evmAddress(txByHash.to);
+      Assertions.evmAddress(receipt.from);
+      Assertions.evmAddress(receipt.to);
+    });
+
+    it('from/to Addresses in transaction to a contract (deployed through the relay) are in evm and long-zero format', async function () {
+      const relayContract = await Utils.deployContractWithEthers([], basicContractJson, accounts[0].wallet, relay);
+
+      const tx = await accounts[0].wallet.sendTransaction({
+        to: relayContract.target,
+        value: ethers.parseEther('1'),
+      });
+
+      await tx.wait();
+
+      let { txByHash, receipt, mirrorResult } = await getTxData(tx.hash);
+
+      mirrorResult.from = accounts[0].wallet.address;
+      mirrorResult.to = relayContract.target;
+
+      Assertions.transaction(txByHash, mirrorResult);
+      Assertions.transactionReceipt(receipt, mirrorResult);
+
+      Assertions.evmAddress(txByHash.from);
+      Assertions.evmAddress(txByHash.to);
+      Assertions.evmAddress(receipt.from);
+      Assertions.evmAddress(receipt.to);
+    });
+
+    it('from/to Addresses in transaction to a contract (deployed through HAPI tx) are in evm and long-zero format', async function () {
+      const tx = await accounts[0].wallet.sendTransaction({
+        to: mirrorContract.evm_address,
+        value: ethers.parseEther('1'),
+      });
+
+      await tx.wait();
+
+      let { txByHash, receipt, mirrorResult } = await getTxData(tx.hash);
+
+      mirrorResult.from = accounts[0].wallet.address;
+      mirrorResult.to = mirrorContract.evm_address;
+
+      Assertions.transaction(txByHash, mirrorResult);
+      Assertions.transactionReceipt(receipt, mirrorResult);
+
+      Assertions.evmAddress(txByHash.from);
+      Assertions.longZeroAddress(txByHash.to);
+      Assertions.evmAddress(receipt.from);
+      Assertions.longZeroAddress(receipt.to);
```

Comment:
From what I've noticed, contracts that have been deployed via the SDK (using `fileCreateTransaction` still show a long-zero address in the mirror node instead of a proper evm address. Here are a couple of the latest contracts on testnet:

- [0.0.1581464](https://testnet.mirrornode.hedera.com/api/v1/contracts/0.0.1581464) - returns long-zero
- [0.0.1581465](https://testnet.mirrornode.hedera.com/api/v1/contracts/0.0.1581465) - returns evm address

Reply:
Very strange.
Can you open an issue on the services side so we can track this nuance.